nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/leader/election/LeaderElectionManager.java

```diff
@@ -40,7 +35,8 @@
     void register(String roleName, LeaderElectionStateChangeListener listener);
 
     /**
-     * Adds a new role for which a leader is required, providing the given value for this node as the Participant ID
+     * Adds a new role for which a leader is required, providing the given value for this node as the Participant ID. If the Participant ID
+     * is <code>null</code>, this node will never be elected leader but will passively observe changes to the leadership.
```

Comment:
@JPercivall that means the node will never been elected leader... unless it later re-registers, providing a Participant ID. I.e., the node can register as an observer and later re-register as a participant.